packages/components/Menu/src/MenuPopover/useMenuPopover.ts

```diff
@@ -53,6 +53,12 @@ export const useMenuPopover = (_props: MenuPopoverProps): MenuPopoverState => {
     setCanFocusOnPopover(false);
   }, [setCanFocusOnPopover]);
 
+  React.useEffect(() => {
+    return function cleanup() {
+      clearTimeout(popoverHoverOutTimer);
+    };
+  });
+
```

Comment:
Could we move the call to `setTimeOut` for the same timer into the same effect? That way, the code related to the popoverOverOutTimer is all in one place?

Got that idea from this talk about when hooks were introduced, where the creator does the same thing -> https://youtu.be/dpw9EHDh2bM?t=2555 (starts at timestamp where he mentions cleanup)

Reply:
Yeah, we could, but since it's in a callback right now I'd have to add some code which would set some state to say "please set a timer on the next rerender cycle" which would then set the timer. It seems kinda convoluted to me. I could move this code closer to the code  which sets the time if that would help?